samples/net/mqtt_publisher/prj_frdm_k64f.conf

```diff
@@ -25,6 +25,7 @@ CONFIG_NET_IPV6=y
 CONFIG_MQTT_LIB=y
 
 CONFIG_NET_APP_SETTINGS=y
+CONFIG_NET_APP_CLIENT=y
```

Comment:
This is not needed as you have that automatically selected when CONFIG_MQTT_LIB is enabled.